frontend/src/metabase/query_builder/containers/QueryBuilder.jsx

```diff
@@ -294,6 +294,10 @@ function QueryBuilder(props) {
     initializeQB(location, params);
   });
 
+  useEffect(() => {
+    initializeQB(location, params);
```

Comment:
This is too eager and breaks lots of things.

1. Open an existing native question (e.g. `select 1;`)
2. Open editor
3. Change query (e.g. `select 1 + 2;`
4. Run question

Expected result: 3
Actual result: "Here's where your results will appear"


https://github.com/metabase/metabase/assets/6830683/29b9bfe7-29dc-4151-a68d-710e52dbdd88

Reply:
Updated the approach to reset UI controls if we hit the back button, and the card has changed. Far less eager haha